col/colserde: TestRecordBatchSerializerDeserializeMemoryEstimate failed

col/colserde.TestRecordBatchSerializerDeserializeMemoryEstimate [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3905007&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3905007&tab=artifacts#/) on master @ [5cd5be83218475ef0cc3c62a856b8d9d22cf142b](https://github.com/cockroachdb/cockroach/commits/5cd5be83218475ef0cc3c62a856b8d9d22cf142b):


```
=== RUN   TestRecordBatchSerializerDeserializeMemoryEstimate
    assertion_compare.go:323: 
        	Error Trace:	record_batch_test.go:352
        	Error:      	"0.33" is not greater than or equal to "0.3841743119266055"
        	Test:       	TestRecordBatchSerializerDeserializeMemoryEstimate
        	Messages:   	[new memory size %d is too far away from original %d 1207 872]
--- FAIL: TestRecordBatchSerializerDeserializeMemoryEstimate (0.00s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRecordBatchSerializerDeserializeMemoryEstimate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

col/colserde.TestRecordBatchSerializerDeserializeMemoryEstimate [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4076170&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4076170&tab=artifacts#/) on master @ [6ff6c77e7aebec2edc947c1ed99a1f7cb67e1a3c](https://github.com/cockroachdb/cockroach/commits/6ff6c77e7aebec2edc947c1ed99a1f7cb67e1a3c):


```
=== RUN   TestRecordBatchSerializerDeserializeMemoryEstimate
    assertion_compare.go:323: 
        	Error Trace:	record_batch_test.go:352
        	Error:      	"0.33" is not greater than or equal to "0.43829113924050633"
        	Test:       	TestRecordBatchSerializerDeserializeMemoryEstimate
        	Messages:   	[new memory size %d is too far away from original %d 1065 1896]
--- FAIL: TestRecordBatchSerializerDeserializeMemoryEstimate (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRecordBatchSerializerDeserializeMemoryEstimate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/col/colserde/record_batch_test.go
+++ b/pkg/col/colserde/record_batch_test.go
@@ -341,12 +341,13 @@ func TestRecordBatchSerializerDeserializeMemoryEstimate(t *testing.T) {
 	newMemorySize := colmem.GetBatchMemSize(dest)
 
 	// We expect that the original and the new memory sizes are relatively close
-	// to each other (do not differ by more than a third). We cannot guarantee
-	// more precise bound here because the capacities of the underlying []byte
-	// slices is unpredictable. However, this check is sufficient to ensure that
-	// we don't double count memory under `Bytes.data`.
+	// to each other, specifically newMemorySize must less than
+	// 4/3*originalMemorySize. We cannot guarantee more precise bound here because
+	// the capacities of the underlying []byte slices is unpredictable. However,
+	// this check is sufficient to ensure that we don't double count memory under
+	// `Bytes.data`.
 	const maxDeviation = float64(0.33)
-	deviation := math.Abs(float64(originalMemorySize-newMemorySize) / (float64(originalMemorySize)))
+	deviation := float64(newMemorySize-originalMemorySize) / float64(originalMemorySize)
 	require.GreaterOrEqualf(t, maxDeviation, deviation,
 		"new memory size %d is too far away from original %d", newMemorySize, originalMemorySize)
 }

